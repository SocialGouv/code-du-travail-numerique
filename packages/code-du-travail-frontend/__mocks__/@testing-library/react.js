/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
