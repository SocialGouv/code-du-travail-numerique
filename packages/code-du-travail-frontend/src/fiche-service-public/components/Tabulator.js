import { Heading, Tabs } from "@socialgouv/cdtn-ui";
import PropTypes from "prop-types";
import React from "react";

import { getText } from "../utils.js";
import { ElementBuilder } from "./ElementBuilder.js";

class Tabulator extends React.PureComponent {
  render() {
    const { data, headingLevel: previousHeadingLevel } = this.props;
    const headingLevel =
      previousHeadingLevel > 0 ? previousHeadingLevel + 1 : 1;

    const tabsData = data.children.map((tab) => {
      return {
        panel: (
          <ElementBuilder
            data={tab.children.filter((el) => el.name !== "Titre")}
            headingLevel={headingLevel}
          />
        ),
        tab: (
          <Heading as={"h" + (headingLevel + 1)}>
            {getText(tab.children.find((child) => child.name === "Titre"))}
          </Heading>
        ),
      };
    });
    return <Tabs data={tabsData} />;
  }
}

Tabulator.propTypes = {
  data: PropTypes.object.isRequired,
  headingLevel: PropTypes.number.isRequired,
};

export default Tabulator;
