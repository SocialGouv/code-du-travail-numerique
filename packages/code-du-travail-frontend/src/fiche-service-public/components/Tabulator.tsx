import React from "react";
import { Tabs } from "@codegouvfr/react-dsfr/Tabs";
import { filterOutTitle, getTitleInChildren } from "../utils";
import { ElementBuilder, FicheSPDataElement } from "./ElementBuilder";
import Title from "./Title";

export const Tabulator = ({
  data,
  headingLevel,
}: {
  data: FicheSPDataElement;
  headingLevel: number;
}) => (
  <Tabs
    tabs={data.children.map((tab: FicheSPDataElement) => {
        const title = getTitleInChildren(tab);
        return {
        label: <Title level={headingLevel}>{title}</Title>,
        content: (
          <ElementBuilder
            data={filterOutTitle(tab)}
            headingLevel={headingLevel + 1}
          />
        ),
      };
    })}
  />
);
export default Tabulator;
