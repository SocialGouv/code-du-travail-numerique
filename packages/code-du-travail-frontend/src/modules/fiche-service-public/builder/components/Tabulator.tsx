"use client";
import React from "react";
import { filterOutTitle, getTitleInChildren } from "../utils";
import { ElementBuilder } from "./ElementBuilder";
import Title from "./Title";
import { FicheSPDataBlocCas, FicheSPDataListeSituations } from "../type";
import { css } from "@styled-system/css";

export const Tabulator = ({
  data,
  headingLevel,
  defaultTabIndex = 0,
}: {
  data: FicheSPDataBlocCas | FicheSPDataListeSituations;
  headingLevel: number;
  defaultTabIndex?: number;
}) => {
  // Prepare tabs data
  const tabs = data.children.map((tab) => {
    const title = getTitleInChildren(tab);
    return {
      label: title,
      content: (
        <>
          <Title className={"fr-sr-only"} level={headingLevel}>
            {title}
          </Title>
          <ElementBuilder
            data={filterOutTitle(tab)}
            headingLevel={headingLevel + 1}
          />
        </>
      ),
    };
  });

  return (
    <div className={css({})}>
      <div className="fr-tabs fr-mb-4w" data-fr-js-tabs="true">
        <ul className="fr-tabs__list" role="tablist">
          {tabs.map((tab, index) => (
            <li role="presentation" key={`tab-${index}`}>
              <button
                id={`tab-${index}`}
                className={`fr-tabs__tab fr-tabs__tab--icon-left ${index === defaultTabIndex ? "fr-tabs__tab--active" : ""}`}
                tabIndex={index === defaultTabIndex ? 0 : -1}
                role="tab"
                aria-selected={index === defaultTabIndex}
                aria-controls={`panel-${index}`}
              >
                {tab.label}
              </button>
            </li>
          ))}
        </ul>

        {tabs.map((tab, index) => (
          <div
            key={`panel-${index}`}
            id={`panel-${index}`}
            className={`fr-tabs__panel ${index === defaultTabIndex ? "fr-tabs__panel--selected" : ""}`}
            role="tabpanel"
            aria-labelledby={`tab-${index}`}
            tabIndex={0}
          >
            {tab.content}
          </div>
        ))}
      </div>
    </div>
  );
};

export default Tabulator;
