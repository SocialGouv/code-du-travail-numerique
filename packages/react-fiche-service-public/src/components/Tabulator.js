import { Heading, Tabs } from "@socialgouv/cdtn-ui";
import PropTypes from "prop-types";
import React from "react";

import { getText } from "../utils.js";
import { ElementBuilder } from "./ElementBuilder.js";

class Tabulator extends React.PureComponent {
  static propTypes = {
    data: PropTypes.object.isRequired,
    headingLevel: PropTypes.number.isRequired,
  };

  render() {
    const { data, headingLevel: previousHeadingLevel } = this.props;
    const headingLevel =
      previousHeadingLevel > 0
        ? previousHeadingLevel + 1
        : previousHeadingLevel;

    const tabsData = data.children.map((tab) => {
      return {
        panel: (
          <ElementBuilder
            data={tab.children.filter((el) => el.name !== "Titre")}
            headingLevel={headingLevel + 1}
          />
        ),
        tab: (
          <Heading
            as={"h" + (headingLevel + 1)}
            stripe="none"
            style={{ margin: 0 }}
            variant="none"
          >
            {getText(tab.children.find((child) => child.name === "Titre"))}
          </Heading>
        ),
      };
    });
    return <Tabs data={tabsData} />;
  }
}

export default Tabulator;
