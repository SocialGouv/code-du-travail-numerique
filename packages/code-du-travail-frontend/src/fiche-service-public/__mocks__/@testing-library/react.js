import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
