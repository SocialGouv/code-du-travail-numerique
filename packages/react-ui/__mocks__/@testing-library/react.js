import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
