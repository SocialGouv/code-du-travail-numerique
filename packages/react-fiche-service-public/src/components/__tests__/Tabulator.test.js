import { render } from "@testing-library/react";
import React from "react";

import Tabulator from "../Tabulator";
import tabulatorDataMock from "./mocks/tabulatorData.json";

describe("<Tabulator />", () => {
  it("should have two different levels of headings", () => {
    const { container } = render(
      <Tabulator data={tabulatorDataMock} headingLevel={0} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should incease heading levels if not 0", () => {
    const { container } = render(
      <Tabulator data={tabulatorDataMock} headingLevel={1} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render", () => {
    const { container } = render(
      <Tabulator data={tabulatorDataMock} headingLevel={2} />
    );
    expect(container).toMatchSnapshot();
  });
});
