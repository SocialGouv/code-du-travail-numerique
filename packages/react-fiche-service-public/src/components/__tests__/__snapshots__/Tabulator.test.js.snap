// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabulator /> should have two different levels of headings 1`] = `
.c8 {
  position: relative;
  display: block;
  margin: 3.2rem 0 2rem 0;
  color: #2f3b6c;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
}

.c7 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: -1px !important;
  padding: 0 !important;
  overflow: hidden !important;
  white-space: nowrap !important;
  border: 0 !important;
  -webkit-clip: rect(0,0,0,0) !important;
  clip: rect(0,0,0,0) !important;
}

.c1 {
  position: relative;
  overflow: hidden;
}

.c1:before,
.c1:after {
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 1;
  display: block;
  width: 4rem;
  opacity: 0;
  -webkit-transition: opacity 0.3s linear;
  transition: opacity 0.3s linear;
  content: "";
  pointer-events: none;
  background: radial-gradient( ellipse at center, #fff 15%, rgba(255,255,255,0) 80% );
}

.c1:before {
  left: -2rem;
}

.c1:after {
  right: -2rem;
}

.c3 {
  overflow-x: auto;
}

.c0 {
  margin-bottom: 3.2rem;
}

.c2 {
  margin-right: 0.4rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  margin: 0;
  padding: 0 0 0 0;
  overflow: visible;
  list-style-type: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-height: 11rem;
  margin-left: 0.4rem;
  padding: 1rem 1.6rem;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.8rem;
  background-color: #fff;
  border: 1px solid #bbcadf;
  border-bottom: 1px solid transparent;
  border-radius: 0.6rem 0.6rem 0 0;
  cursor: pointer;
  opacity: 1;
  -webkit-transition: opacity 250ms linear;
  transition: opacity 250ms linear;
}

.c5[aria-selected="false"]:hover {
  opacity: 0.7;
}

.c5[aria-selected="true"] {
  color: #fff;
  background-color: #7994d4;
}

.c6 {
  color: #3e486e;
  background-color: #fff;
}

.c6.react-tabs__tab-panel--selected {
  padding: 2.4rem;
  border: 1px solid #bbcadf;
  border-radius: 0.6rem;
}

.c9 > *:first-child {
  margin-top: 0;
}

.c9 > *:last-child {
  margin-bottom: 0;
}

@media (max-width:600px) {
  .c8 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c1 {
    overflow-x: hidden;
  }
}

@media (max-width:600px) {
  .c1:before,
  .c1:after {
    display: block;
  }
}

@media (max-width:600px) {
  .c3 {
    overflow-x: auto;
  }
}

@media (max-width:600px) {
  .c0 {
    margin-bottom: 2rem;
  }
}

@media (max-width:980px) {
  .c4 {
    padding: 0;
  }
}

@media (max-width:980px) {
  .c5 {
    margin: 0.4rem;
    border-bottom: 1px solid #bbcadf;
    border-radius: 0.6rem;
  }
}

@media (max-width:600px) {
  .c5 {
    font-size: 1.6rem;
    max-width: 150px;
    -webkit-flex: none !important;
    -ms-flex: none !important;
    flex: none !important;
  }
}

@media (max-width:980px) {
  .c6.react-tabs__tab-panel--selected {
    margin-top: 0.4rem;
  }
}

@media (max-width:600px) {
  .c6.react-tabs__tab-panel--selected {
    padding: 1rem;
  }
}

<div>
  <div
    class="c0"
    data-tabs="true"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <ul
          class="c4"
          role="tablist"
        >
          <li
            aria-controls="react-tabs-1"
            aria-disabled="false"
            aria-selected="true"
            class="c5 react-tabs__tab--selected"
            data-testid="undefined-0"
            id="react-tabs-0"
            role="tab"
            tabindex="0"
          >
            CDI
          </li>
          <li
            aria-controls="react-tabs-3"
            aria-disabled="false"
            aria-selected="false"
            class="c5"
            data-testid="undefined-1"
            id="react-tabs-2"
            role="tab"
          >
            CDD
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-0"
      class="c6 react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      <div
        class="c7"
      >
        <h2
          class="c8"
          data-testid="heading"
        >
          CDI
        </h2>
      </div>
      <div
        class="c9"
      >
        <h3
          class="c8"
          data-testid="heading"
        >
          titre 1
        </h3>
        <p>
          paragraphe 1
        </p>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="c6"
      id="react-tabs-3"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`<Tabulator /> should incease heading levels if not 0 1`] = `
.c8 {
  position: relative;
  display: block;
  margin: 3.2rem 0 2rem 0;
  color: #2f3b6c;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
}

.c7 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: -1px !important;
  padding: 0 !important;
  overflow: hidden !important;
  white-space: nowrap !important;
  border: 0 !important;
  -webkit-clip: rect(0,0,0,0) !important;
  clip: rect(0,0,0,0) !important;
}

.c1 {
  position: relative;
  overflow: hidden;
}

.c1:before,
.c1:after {
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 1;
  display: block;
  width: 4rem;
  opacity: 0;
  -webkit-transition: opacity 0.3s linear;
  transition: opacity 0.3s linear;
  content: "";
  pointer-events: none;
  background: radial-gradient( ellipse at center, #fff 15%, rgba(255,255,255,0) 80% );
}

.c1:before {
  left: -2rem;
}

.c1:after {
  right: -2rem;
}

.c3 {
  overflow-x: auto;
}

.c0 {
  margin-bottom: 3.2rem;
}

.c2 {
  margin-right: 0.4rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  margin: 0;
  padding: 0 0 0 0;
  overflow: visible;
  list-style-type: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-height: 11rem;
  margin-left: 0.4rem;
  padding: 1rem 1.6rem;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.8rem;
  background-color: #fff;
  border: 1px solid #bbcadf;
  border-bottom: 1px solid transparent;
  border-radius: 0.6rem 0.6rem 0 0;
  cursor: pointer;
  opacity: 1;
  -webkit-transition: opacity 250ms linear;
  transition: opacity 250ms linear;
}

.c5[aria-selected="false"]:hover {
  opacity: 0.7;
}

.c5[aria-selected="true"] {
  color: #fff;
  background-color: #7994d4;
}

.c6 {
  color: #3e486e;
  background-color: #fff;
}

.c6.react-tabs__tab-panel--selected {
  padding: 2.4rem;
  border: 1px solid #bbcadf;
  border-radius: 0.6rem;
}

.c9 > *:first-child {
  margin-top: 0;
}

.c9 > *:last-child {
  margin-bottom: 0;
}

@media (max-width:600px) {
  .c8 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c1 {
    overflow-x: hidden;
  }
}

@media (max-width:600px) {
  .c1:before,
  .c1:after {
    display: block;
  }
}

@media (max-width:600px) {
  .c3 {
    overflow-x: auto;
  }
}

@media (max-width:600px) {
  .c0 {
    margin-bottom: 2rem;
  }
}

@media (max-width:980px) {
  .c4 {
    padding: 0;
  }
}

@media (max-width:980px) {
  .c5 {
    margin: 0.4rem;
    border-bottom: 1px solid #bbcadf;
    border-radius: 0.6rem;
  }
}

@media (max-width:600px) {
  .c5 {
    font-size: 1.6rem;
    max-width: 150px;
    -webkit-flex: none !important;
    -ms-flex: none !important;
    flex: none !important;
  }
}

@media (max-width:980px) {
  .c6.react-tabs__tab-panel--selected {
    margin-top: 0.4rem;
  }
}

@media (max-width:600px) {
  .c6.react-tabs__tab-panel--selected {
    padding: 1rem;
  }
}

<div>
  <div
    class="c0"
    data-tabs="true"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <ul
          class="c4"
          role="tablist"
        >
          <li
            aria-controls="react-tabs-5"
            aria-disabled="false"
            aria-selected="true"
            class="c5 react-tabs__tab--selected"
            data-testid="undefined-0"
            id="react-tabs-4"
            role="tab"
            tabindex="0"
          >
            CDI
          </li>
          <li
            aria-controls="react-tabs-7"
            aria-disabled="false"
            aria-selected="false"
            class="c5"
            data-testid="undefined-1"
            id="react-tabs-6"
            role="tab"
          >
            CDD
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-4"
      class="c6 react-tabs__tab-panel--selected"
      id="react-tabs-5"
      role="tabpanel"
    >
      <div
        class="c7"
      >
        <h3
          class="c8"
          data-testid="heading"
        >
          CDI
        </h3>
      </div>
      <div
        class="c9"
      >
        <h4
          class="c8"
          data-testid="heading"
        >
          titre 1
        </h4>
        <p>
          paragraphe 1
        </p>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-6"
      class="c6"
      id="react-tabs-7"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`<Tabulator /> should render 1`] = `
.c8 {
  position: relative;
  display: block;
  margin: 3.2rem 0 2rem 0;
  color: #2f3b6c;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
}

.c7 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: -1px !important;
  padding: 0 !important;
  overflow: hidden !important;
  white-space: nowrap !important;
  border: 0 !important;
  -webkit-clip: rect(0,0,0,0) !important;
  clip: rect(0,0,0,0) !important;
}

.c1 {
  position: relative;
  overflow: hidden;
}

.c1:before,
.c1:after {
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 1;
  display: block;
  width: 4rem;
  opacity: 0;
  -webkit-transition: opacity 0.3s linear;
  transition: opacity 0.3s linear;
  content: "";
  pointer-events: none;
  background: radial-gradient( ellipse at center, #fff 15%, rgba(255,255,255,0) 80% );
}

.c1:before {
  left: -2rem;
}

.c1:after {
  right: -2rem;
}

.c3 {
  overflow-x: auto;
}

.c0 {
  margin-bottom: 3.2rem;
}

.c2 {
  margin-right: 0.4rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  margin: 0;
  padding: 0 0 0 0;
  overflow: visible;
  list-style-type: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-height: 11rem;
  margin-left: 0.4rem;
  padding: 1rem 1.6rem;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.8rem;
  background-color: #fff;
  border: 1px solid #bbcadf;
  border-bottom: 1px solid transparent;
  border-radius: 0.6rem 0.6rem 0 0;
  cursor: pointer;
  opacity: 1;
  -webkit-transition: opacity 250ms linear;
  transition: opacity 250ms linear;
}

.c5[aria-selected="false"]:hover {
  opacity: 0.7;
}

.c5[aria-selected="true"] {
  color: #fff;
  background-color: #7994d4;
}

.c6 {
  color: #3e486e;
  background-color: #fff;
}

.c6.react-tabs__tab-panel--selected {
  padding: 2.4rem;
  border: 1px solid #bbcadf;
  border-radius: 0.6rem;
}

.c9 > *:first-child {
  margin-top: 0;
}

.c9 > *:last-child {
  margin-bottom: 0;
}

@media (max-width:600px) {
  .c8 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c1 {
    overflow-x: hidden;
  }
}

@media (max-width:600px) {
  .c1:before,
  .c1:after {
    display: block;
  }
}

@media (max-width:600px) {
  .c3 {
    overflow-x: auto;
  }
}

@media (max-width:600px) {
  .c0 {
    margin-bottom: 2rem;
  }
}

@media (max-width:980px) {
  .c4 {
    padding: 0;
  }
}

@media (max-width:980px) {
  .c5 {
    margin: 0.4rem;
    border-bottom: 1px solid #bbcadf;
    border-radius: 0.6rem;
  }
}

@media (max-width:600px) {
  .c5 {
    font-size: 1.6rem;
    max-width: 150px;
    -webkit-flex: none !important;
    -ms-flex: none !important;
    flex: none !important;
  }
}

@media (max-width:980px) {
  .c6.react-tabs__tab-panel--selected {
    margin-top: 0.4rem;
  }
}

@media (max-width:600px) {
  .c6.react-tabs__tab-panel--selected {
    padding: 1rem;
  }
}

<div>
  <div
    class="c0"
    data-tabs="true"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <ul
          class="c4"
          role="tablist"
        >
          <li
            aria-controls="react-tabs-9"
            aria-disabled="false"
            aria-selected="true"
            class="c5 react-tabs__tab--selected"
            data-testid="undefined-0"
            id="react-tabs-8"
            role="tab"
            tabindex="0"
          >
            CDI
          </li>
          <li
            aria-controls="react-tabs-11"
            aria-disabled="false"
            aria-selected="false"
            class="c5"
            data-testid="undefined-1"
            id="react-tabs-10"
            role="tab"
          >
            CDD
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-8"
      class="c6 react-tabs__tab-panel--selected"
      id="react-tabs-9"
      role="tabpanel"
    >
      <div
        class="c7"
      >
        <h4
          class="c8"
          data-testid="heading"
        >
          CDI
        </h4>
      </div>
      <div
        class="c9"
      >
        <h5
          class="c8"
          data-testid="heading"
        >
          titre 1
        </h5>
        <p>
          paragraphe 1
        </p>
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-10"
      class="c6"
      id="react-tabs-11"
      role="tabpanel"
    />
  </div>
</div>
`;
